reframe/core/schedulers/oar.py

```diff
@@ -0,0 +1,201 @@
+        )
+
+    def emit_preamble(self, job):
+        # Same reason as oarsub, we give full path to output and error files to
+        # avoid writing them in the working dir
+        preamble = [
+            self._format_option('-n "%s"' % job.name),
+            self._format_option('-O %s' %
+                                os.path.join(job.workdir, job.stdout)),
+            self._format_option('-E %s' %
+                                os.path.join(job.workdir, job.stderr)),
+        ]
+
+        if job.time_limit is not None:
+            h, m, s = seconds_to_hms(job.time_limit)
+            self.TASKS_OPT += ',walltime=%d:%d:%d' % (h, m, s)
+
+        # Get number of nodes in the reservation
+        num_tasks_per_node = job.num_tasks_per_node or 1
+        num_nodes = job.num_tasks // num_tasks_per_node
+
+        # Emit main resource reservation option
+        options = [self.TASKS_OPT.format(
+            num_nodes=num_nodes, num_tasks_per_node=num_tasks_per_node,
+        )]
+
+        # Emit the rest of the options
+        options += job.sched_access + job.options + job.cli_options
+
+        for opt in options:
+            if opt.startswith('#'):
+                preamble.append(opt)
+            else:
+                preamble.append(self._format_option(opt))
+
+        # OAR starts the job in the home directory by default
+        preamble.append('cd %s' % job.workdir)
+        return preamble
+
+    def submit(self, job):
+        # For some reason OAR job manager says that job launching dir is
+        # working dir of the repo and not stage dir. A workaround is to give
+        # full path of script to oarsub
+        job_script_fullpath = os.path.join(job.workdir, job.script_filename)
+
+        # OAR needs -S to submit job in batch mode
+        cmd = f'oarsub -S {job_script_fullpath}'
+        completed = _run_strict(cmd, timeout=self._submit_timeout)
+        jobid_match = re.search(r'.*OAR_JOB_ID=(?P<jobid>\S+)',
+                                completed.stdout)
+        if not jobid_match:
+            raise JobSchedulerError('could not retrieve the job id '
+                                    'of the submitted job')
+
+        job._jobid = jobid_match.group('jobid')
+        job._submit_time = time.time()
+
+    def cancel(self, job):
+        _run_strict(f'oardel {job.jobid}', timeout=self._submit_timeout)
+        job._cancelled = True
+
+    def poll(self, *jobs):
+        if jobs:
+            # Filter out non-jobs
+            jobs = [job for job in jobs if job is not None]
+
+        if not jobs:
+            return
+
+        for job in jobs:
+            completed = osext.run_command(
+                f'oarstat -fj {job.jobid}'
+            )
+
+            if completed.returncode != 0:
+                raise JobSchedulerError(
+                    f'oarstat failed with exit code {completed.returncode} '
+                    f'(standard error follows):\n{completed.stderr}'
+                )
```

Comment:
Just use `_run_strict()` here and do not throw an extra error.

Reply:
Fixed